tour_model_eval/evaluation_pipeline.py

```diff
@@ -0,0 +1,173 @@
+import emission.analysis.modelling.tour_model.similarity as similarity
+import numpy as np
+import emission.analysis.modelling.tour_model.get_request_percentage as grp
+import emission.analysis.modelling.tour_model.get_scores as gs
+import emission.analysis.modelling.tour_model.label_processing as lp
+import emission.analysis.modelling.tour_model.data_preprocessing as preprocess
+
+def second_round(first_label_set,first_labels,bin_trips,filter_trips,low,dist_pct,sim,new_labels,track):
```

Comment:
do you really need 9 mandatory arguments to this function? What do you need `sim` for, for example, if you already have `filter_trips` and `track`?

do you really need both `first_label_set` and `first_labels`? Why can you not get `first_label_set` from `first_label` in this function?

At a high level, if you get to so many arguments, I will recommend that you create a class to pass them in. But given our time constraints, I am not going to push for this now.

Reply:
`sim` is to find out trips below the cutoff in the 1st round. Those trips didn't go through further analysis except the 1st round, so I have to use `sim`

> At a high level, if you get to so many arguments, I will recommend that you create a class to pass them in. But given our time constraints, I am not going to push for this now.

I don't understand this. Maybe I can reduce 1 or 2 arguments. But given that there are still many arguments I have to pass in, what is the difference between writing a pure function to accept them and writing a class?

Reply:
you can write a class/struct to encapsulate them so that people don't have to write really long lines passing them back and forth. You can have a single class that you both pass in **and return** and that has the current state of the algorithm. Makes everything a lot less verbose.

Reply:
I have created a class for the second round. See `second_round_of_clustering.py`
Although it is not a perfect one, but it works.